docs/pages/fund-apps/estimate-fees.mdx

```diff
@@ -0,0 +1,128 @@
+2.  **XMTP Appchain** (onchain): A specialized blockchain that stores critical metadata, such as user identities, contacts, and group permissions. This onchain layer is essential for decentralized identity and ensuring that only the appropriate users can access conversations.
+
+### Messaging fees
+
+Messages sent through the **XMTP Broadcast Network** incur these types of **messaging fees**:
+
+- **Base fee**: A flat per-message fee.
+- **Storage fee**: A fee charged per-byte-day of storage required. A 100-byte message stored for 30 days would be calculated as 3,000-byte-days.
+- **Congestion fee**: A dynamic fee computed by looking at the recent activity of an originator. Added only during periods of high network activity.
+
+The dollar value of the flat per-message base fee, per-byte-day storage fee, and congestion fee are stored in a smart contract and remain constant for a specified period of time. These fees are set and adjusted through protocol governance.
+
+Every message an app or agent sends through the XMTP Broadcast Network counts toward its message volume. These message types include:
+
+- App messages (text, reactions, replies)
+- Media attachments (charged by size, capped at 1 MB)
+- System messages (read receipts, typing indicators)
+
+To learn more, see [Envelope types](/protocol/envelope-types).
+
+Collected messaging fees are paid directly to the node operators who run the globally distributed nodes that power the XMTP Broadcast Network.
+
+#### Messaging fee payment flow
+
+This sequence diagram illustrates the messaging fee payment flow from payers to independent node operators using a series of smart contracts.
+
+<Zoom>
+<Mermaid
+  id="payer-to-node"
+  chart={`
+    sequenceDiagram
+        actor Payer
+        participant PayerRegistry
+        participant PayerReportManager
+        participant DistributionManager
+        participant NodeRegistry
+        actor NodeOperator as Node operator
+
+        Payer->>PayerRegistry: 1. Allocates USDC to cover usage fees
+        Note over PayerReportManager: Node operator submits payer usage reports
+        PayerReportManager->>PayerRegistry: 2. Deducts fees from payer's balance
+        PayerRegistry->>DistributionManager: 3. Transfers collected fees
+        NodeOperator->>DistributionManager: 4. Claims share of fees
+        DistributionManager->>NodeRegistry: 5. Verifies claimant is node owner
+        NodeRegistry-->>DistributionManager: Returns node owner's address
+        DistributionManager->>NodeOperator: 6. Transfers earned fees to node operator
+`}
+/>
+</Zoom>
+
+### Gas fees
+
+Commit messages sent through the **XMTP Appchain** incur **gas fees**. Commit messages require transactions on the XMTP Appchain. Gas fees are paid directly from the payer wallet's XMTP Appchain allowance.
+
+Every commit message an app or agent sends through the XMTP Appchain counts toward its message volume. Commit messages are typically used to perform:
```

Comment:
Transactions on the appchain are not messages and do not draw down the PayerRegistry balance. Right @deluca-mike ?